docs/gno-infra/setting-up-a-faucet.md

```diff
@@ -0,0 +1,173 @@
+---
+id: setting-up-a-faucet
```

Comment:
This tutorial doesn't cover:

- `gnofaucet`
- `gnofaucet` vs `gnolang/faucet`
- different things I've noted as part of the comments

Reply:
Looking at the original issue, this tutorial was meant to cover setting up a faucet with `gnolang/faucet`, which it does.

I suggest two options:
1. We keep the scope of this PR to cover `gnolang/faucet`, and open a separate issue for documenting `gnofaucet`. Ideally, we would have some reference docs or at least a basic README about it before I start working on it.
2. I'm unfamiliar with the `gnofaucet` functionalities and codebase - understanding the details of it are a requirement for me to be able to write a high-quality tutorial. Possibly we can jump on a quick 15 min call and you can run me through the main functionalities to speed up the process, and get this done asap?

I have fixed up all of the other comments as requested, and added a section mentioning that the faucet can be extended and referenced `gnofaucet` as an example: [d2388e7](https://github.com/gnolang/gno/pull/2122/commits/d2388e76abe1eb5a254290ee6d252a3c52eb2146)

WDYT?

cc @Kouteki

Reply:
I like option 1 better - it lets us publish this PR and make incremental progress.

Reply:
@leohhhn 

We discussed on a call that `gnofaucet` is most definitely in the scope of this PR, and should be added -- please add it and ping me when it's up in the PR 🙏

Reply:
Frankly speaking, I do not remember discussing this distinction on a call. 

However, I did take about 2h to read through the code of `gnofaucet` and write up a section on it. Please check it out: [451c0a9](https://github.com/gnolang/gno/pull/2122/commits/451c0a90aa3b53c4b25be6d9eb3e7cc27e4a7aed)